internal/geturl/geturl.go

```diff
@@ -0,0 +1,67 @@
+package geturl
+
+import (
+	"io"
+	"net/http"
+
+	"github.com/cenkalti/backoff/v3"
+	"github.com/taskcluster/httpbackoff/v3"
+)
+
+type HTTPRetryError struct {
+	Attempts int
+	Err      error
+}
```

Comment:
User code that imports the taskcluster client will often benefit from being able to inspect a non-nil error that is returned from a client method. If calling code receives an error type that comes from an internal package, will it still be able to access the internal fields of the error? Perhaps HTTPRetryError needs to be a public type for calling code to be able to access the underlying error?

Reply:
It's re-exported in the download package.

Reply:
*in the tcobject package